include/openthread/nat64.h

```diff
@@ -87,6 +87,108 @@ typedef struct otIp4Cidr
     uint8_t      mLength;
 } otIp4Cidr;
 
+typedef struct otIp4Message otIp4Message;
+
+/**
+ * Allocate a new message buffer for sending an IPv4 message (which will be translated into an IPv6 packet by NAT64
+ * later). Message buffers allocated by this function will have 20 bytes (The differences between the size of IPv6
+ * headers and the size of IPv4 headers) reserved.
+ *
+ * This function is available only when OPENTHREAD_CONFIG_NAT64_TRANSLATOR_ENABLE is enabled.
+ *
+ * @note If @p aSettings is 'NULL', the link layer security is enabled and the message priority is set to
+ * OT_MESSAGE_PRIORITY_NORMAL by default.
+ *
+ * @param[in]  aInstance  A pointer to an OpenThread instance.
+ * @param[in]  aSettings  A pointer to the message settings or NULL to set default settings.
+ *
+ * @returns A pointer to the message buffer or NULL if no message buffers are available or parameters are invalid.
+ *
+ * @sa otIp4MessageFree
+ *
+ */
+otIp4Message *otIp4NewMessage(otInstance *aInstance, const otMessageSettings *aSettings);
+
+/**
+ * Free an allocated IPv4 message buffer.
+ *
+ * @param[in]  aMessage  A pointer to a message buffer.
+ *
+ */
+void otIp4MessageFree(otIp4Message *aMessage);
```

Comment:
Is this necessary? Can we share https://github.com/openthread/openthread/blob/main/include/openthread/message.h#L96?